contracts/script/multisigTransactionProposals/safeSDK/modifyStateHistoryRetentionPeriod.ts

```diff
@@ -0,0 +1,130 @@
+    const safeAddress = getEnvVar("SAFE_MULTISIG_ADDRESS");
+    validateEthereumAddress(safeAddress);
+    const safeSdk = await Safe.create({ ethAdapter, safeAddress });
+    const orchestratorSignerAddress = await orchestratorSigner.getAddress();
+
+    const stateHistoryRetentionPeriod = parseInt(getEnvVar("STATE_HISTORY_RETENTION_PERIOD"));
+    console.log("Modify the stateHistoryRetentionPeriod to ", stateHistoryRetentionPeriod);
+
+    await proposeSetStateHistoryRetentionTransaction(
+      safeSdk,
+      safeService,
+      orchestratorSignerAddress,
+      safeAddress,
+      stateHistoryRetentionPeriod,
+    );
+
+    console.log(
+      `The other owners of the Safe Multisig wallet need to sign the transaction via the Safe UI https://app.safe.global/transactions/queue?safe=sep:${safeAddress}`,
+    );
+  } catch (error) {
+    throw new Error("An error occurred: " + error);
+  }
+}
+
+/**
+ * Function to propose the transaction data for setting the state history retention period
+ * @param {string} safeSDK - An instance of the Safe SDK
+ * @param {string} safeService - An instance of the Safe Service
+ * @param {string} signerAddress - The address of the address signing the transaction
+ * @param {string} safeAddress - The address of the Safe multisig wallet
+ * @param {number} stateHistoryRetentionPeriod - The state history retention period in seconds
+ */
+export async function proposeSetStateHistoryRetentionTransaction(
+  safeSDK: Safe,
+  safeService: SafeApiKit,
+  signerAddress: string,
+  safeAddress: string,
+  stateHistoryRetentionPeriod: number,
+) {
+  // Prepare the transaction data to set the stateHistoryRetentionPeriod (seconds)
+  let data = createStateHistoryRetentionPeriodTxData(stateHistoryRetentionPeriod);
+
+  const contractAddress = getEnvVar("LIGHT_CLIENT_CONTRACT_PROXY_ADDRESS");
+  validateEthereumAddress(contractAddress);
+
+  // Create the Safe Transaction Object
+  const safeTransaction = await createSafeTransaction(safeSDK, contractAddress, data, "0");
+
+  // Get the transaction hash and sign the transaction
+  const safeTxHash = await safeSDK.getTransactionHash(safeTransaction);
+
+  // Sign the transaction with orchestrator signer that was specified when we created the safeSDK
+  const senderSignature = await safeSDK.signHash(safeTxHash);
+
+  // Propose the transaction which can be signed by other owners via the Safe UI
+  await safeService.proposeTransaction({
+    safeAddress: safeAddress,
+    safeTransactionData: safeTransaction.data,
+    safeTxHash: safeTxHash,
+    senderAddress: signerAddress,
+    senderSignature: senderSignature.data,
+  });
+}
+
+/**
+ * Function to create the transaction data for setting the stateHistoryRetentionPeriod
+ * @param {number} retention_period - The state history retention period in seconds
+ * @returns {string} - Encoded transaction data
+ */
+function createStateHistoryRetentionPeriodTxData(retention_period: number): string {
```

Comment:
nit: does this have to be a function? it is just two lines so we can prob merge this with `proposeSetStateHistoryRetentionTransaction`